src/main/scala/nl/biopet/tools/tenxkit/mergebams/MergeBams.scala

```diff
@@ -66,10 +69,18 @@ object MergeBams extends ToolCommand[Args] {
       new SAMFileWriterFactory().makeBAMWriter(header, true, cmdArgs.outputBam)
 
     val it = new PrefixIterator(bamReaders, dict, cmdArgs.sampleTag)
-    it.foreach(writer.addAlignment)
+    it.foreach { record =>
+      if (cmdArgs.downsampleFraction >= 1.0) writer.addAlignment(record)
+      else {
+        if ((random
+              .nextInt(1000)
+              .toDouble / 1000) <= cmdArgs.downsampleFraction)
```

Comment:
using `random.nextDouble` is much more appropriate here given cmdArgs.downsampleFraction is given as a double. Using `nextInt` and converting to a Double precision number later is a very roundabout way of doing things.

Reply:
that's because I need a specific range. There is a method for a range for doubles, only 0 until something for ints